### ISSUE TITLE
Relative import error

### ISSUE THREAD

**Issue by Ivoah** [micropython/micropython#1114]
I have a directory structure like this:
```
test/
|___ __init__.py
\___ x
    |___ __init__.py
    \___ y.py
```

test/\__init\__.py has:
```python
from .x import *
print('init')
```

test/x/\__init\__.py is:
```python
from .y import *
print('x')
```

and test/x/y.py is:
```python
print('y')
```

When I run `import test` in CPython the output is:
```
y
x
init
```

While micropython throws an error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "test/__init__.py", line 1, in <module>
  File "test/x/__init__.py", line 1, in <module>
ImportError: no module named 'test.y'
```

Micropython tries to import test.y, while it should be importing test.x.y

**Comment by dpgeorge** [comment]
@pfalcon will you have time to look at this or should I have a go?

**Comment by pfalcon** [comment]
I'll see if I can look into it during today's flights. If not, I'd treat with lower priority (workaround is known).

**Comment by pfalcon** [comment]
Fixed in 9e6c82960e38963e4392ad3180811d10657519b3

**Comment by dpgeorge** [comment]
Thanks!
